api/categorisation.go

```diff
@@ -5,18 +5,19 @@ import (
 
 	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
 	"github.com/ONSdigital/dp-api-clients-go/v2/cantabularmetadata"
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/model"
 	"github.com/pkg/errors"
 )
 
-func (api *API) CheckDefaultCategorisation(dimName string, datasetName string) (string, string, error) {
+func (api *API) CheckDefaultCategorisation(dimension *model.Dimension, datasetName string) (string, string, string, error) {
```

Comment:
I can't really work out what this function is supposed to be achieving without following the code letter by letter. This could probably be improved with better naming and perhaps refactoring/splitting it up appropriately.

Reply:
Added a comment and also changed the function name to make a little bit more intuitve sense